From 6958753113ec78ee3b272cd5e4ac68657a0b967a Mon Sep 17 00:00:00 2001
From: Josh Fazekas <Josh.Fazekas@Intel.com>
Date: Mon, 19 May 2025 23:19:22 +0300
Subject: [PATCH] [SW-184431] persist_info: calculate max RMA size

Add max size macro for each table and the total size to be used by
callers of rma_table_init

Removed unused and incorrect size macros from c file

Change-Id: I198414ee03b51ed030866598490b7be98a48e132
Signed-off-by: Josh Fazekas <Josh.Fazekas@Intel.com>
---

diff --git a/persist_info.c b/persist_info.c
index d4f5037..08b92f5 100644
--- a/persist_info.c
+++ b/persist_info.c
@@ -10,7 +10,6 @@
 #include <rma_if.h>
 #include <string.h>
 #include <rma_if.h>
-#include <rma_if_dev.h>
 #include <persist_info.h>
 #include <zephyr/device.h>
 #include <zephyr/kernel.h>
@@ -25,27 +24,6 @@
 #define FLASH_SECTOR_MAGIC		0xAA55AA55
 #define FLASH_HEADER_SIZE		8
 #define ROUND_UP4(n)			(((n) + sizeof(uint32_t) - 1) & ~(sizeof(uint32_t) - 1))
-#define RMA_TBL_HDR_SZ			sizeof(struct table_header)
-#define RMA_IND_SZ			sizeof(struct rma_entry)
-#define RMA_SRAM_ECC_SZ			sizeof(struct rma_sram_ecc_entry)
-#define RMA_HBM_ECC_SZ			sizeof(struct rma_hbm_ecc_entry)
-#define RMA_PARITY_SZ			sizeof(struct rma_hbm_parity_entry)
-#define RMA_ROW_REP_SZ			sizeof(struct rma_hbm_row_replacement_entry)
-#define RMA_PAGE_DISCARD_SZ		sizeof(struct rma_hbm_ecc_entry)
-#define RMA_SPARE_ROW_SZ		sizeof(struct rma_hbm_spare_row_entry)
-#define RMA_MAX_SRAM_ECC_TBL		(RMA_SRAM_ECC_SZ * MAX_SRAM_ECC_ENTRIES + RMA_TBL_HDR_SZ)
-#define RMA_MAX_HBM_ECC_TBL		(RMA_HBM_ECC_SZ * MAX_HBM_ECC_ENTRIES + RMA_TBL_HDR_SZ)
-#define RMA_MAX_PARITY_ECC_TBL		(RMA_PARITY_SZ * MAX_PARITY_ENTRIES + RMA_TBL_HDR_SZ)
-#define RMA_MAX_ROW_REP_TBL		(RMA_ROW_REP_SZ * MAX_ROW_REPLACE_ENTRIES + RMA_TBL_HDR_SZ)
-#define RMA_MAX_RMA_IND_TBL		(RMA_IND_SZ + RMA_TBL_HDR_SZ)
-#define RMA_MAX_PAGE_DISCARD_TBL	(RMA_PAGE_DISCARD_SZ * MAX_PAGE_DISCARDS_ENTRIES +\
-					 RMA_TBL_HDR_SZ)
-#define RMA_MAX_SPARE_ROW_TBL		(RMA_SPARE_ROW_SZ * MAX_HBM_SPARE_ROWS_ENTRIES +\
-					 RMA_TBL_HDR_SZ)
-#define MAX_TOTAL_SZ			(RMA_MAX_SRAM_ECC_TBL + RMA_MAX_HBM_ECC_TBL +\
-					 RMA_MAX_PARITY_ECC_TBL +\
-					 RMA_MAX_ROW_REP_TBL + RMA_MAX_RMA_IND_TBL +\
-					 RMA_MAX_PAGE_DISCARD_TBL + RMA_MAX_SPARE_ROW_TBL)
 #define WAIT_TIMEOUT			4000
 #define FLASH_NUM_FLASH_INFO		2
 #define FLASH_SECTOR_MAGIC		0xAA55AA55
diff --git a/persist_info.h b/persist_info.h
index e1f8d92..69857d2 100644
--- a/persist_info.h
+++ b/persist_info.h
@@ -10,6 +10,28 @@
 
 #include <zephyr/device.h>
 #include <rma_if.h>
+#include <rma_if_dev.h>
+
+#define RMA_MAX_RMA_HDR_TBL_SZ		(sizeof(struct rma_entry) *\
+					 MAX_RMA_HEADER_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_SRAM_ECC_TBL_SZ		(sizeof(struct rma_sram_ecc_entry) *\
+					 MAX_SRAM_ECC_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_HBM_ECC_TBL_SZ		(sizeof(struct rma_hbm_ecc_entry) *\
+					 MAX_HBM_ECC_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_HBM_PARITY_TBL_SZ	(sizeof(struct rma_hbm_parity_entry) *\
+					 MAX_PARITY_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_ROW_REP_TBL_SZ		(sizeof(struct rma_hbm_row_replacement_entry) *\
+					 MAX_ROW_REPLACE_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_PAGE_DISCARD_TBL_SZ	(sizeof(struct rma_hbm_row_replacement_entry) *\
+					 MAX_PAGE_DISCARDS_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_HBM_LANE_REPAIR_TBL_SZ	(sizeof(struct rma_hbm_lane_repair_entry) *\
+					 MAX_HBM_LANE_REPAIR_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_SPARE_ROW_TBL_SZ	(sizeof(struct rma_hbm_spare_row_entry) *\
+					 MAX_HBM_SPARE_ROWS_ENTRIES + sizeof(struct table_header))
+#define RMA_MAX_TOTAL_SZ	(RMA_MAX_RMA_HDR_TBL_SZ + RMA_MAX_SRAM_ECC_TBL_SZ +\
+				 RMA_MAX_HBM_ECC_TBL_SZ + RMA_MAX_HBM_PARITY_TBL_SZ +\
+				 RMA_MAX_ROW_REP_TBL_SZ + RMA_MAX_PAGE_DISCARD_TBL_SZ +\
+				 RMA_MAX_HBM_LANE_REPAIR_TBL_SZ + RMA_MAX_SPARE_ROW_TBL_SZ)
 
 enum persist_info_error {
 	PERSIST_INFO_ERR_NONE,
